core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -205,7 +205,9 @@ protected void doWriteTo(StreamOutput out) throws IOException {
         out.writeOptionalString(minimumShouldMatch);
         if (out.getVersion().onOrAfter(Version.V_5_1_1)) {
             out.writeOptionalString(settings.quoteFieldSuffix());
-            out.writeOptionalBoolean(useAllFields);
+            if (out.getVersion().before(Version.V_6_0_0_beta2)) {
+                out.writeOptionalBoolean(null);
```

Comment:
I believe this does need to calculate whether all fields mode is going to be used, otherwise a mixed 5.6 - 6.0 cluster will have the behavior correct on 6.0 and incorrect on 5.6, since 5.6 calculates the decision for whether to use `all_fields` using different criteria than 6.0 will?